Assets/Scripts/Puzzles/Color/UI/ColorCommandDropzone.cs

```diff
@@ -0,0 +1,45 @@
+﻿using DN.UI;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace DN.Puzzle.Color
+{
+	/// <summary>
+	/// ADD CLASS SUMMARY!
+	/// </summary>
+	public class ColorCommandDropzone : MonoBehaviour, IDroppable
+	{
+		private struct Item
+		{
+			public DraggableItem draggableItem;
+			public ColorCommand colorCommand;
+			public Transform previousParent;
+		}
+
+		public IEnumerable<ColorCommand> CommandQueue => colors.Select(x=>x.colorCommand);
+
+		private List<Item> colors = new List<Item>();
+
+		public void Drop(DraggableItem droppedObject)
+		{
+			colors.Add(new Item
+			{
+				draggableItem = droppedObject,
+				previousParent = droppedObject.transform.parent,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'droppedObject' before using it.](https://app.codacy.com/manual/stepperman/pvb-programming/pullRequest?prid=5403921)